### ISSUE TITLE
DOR Remake Net Test crashes on start up

### ISSUE THREAD

**Issue by SpacePirate2k** [suicvne/DOR_Remake#1]
Version: Alpha Version 0.6.8.0_nettest
Action: Executing ''DOR Net Test'' .exe
OS: Windows 10 20H2
Specs: Intel Core i7-1065G7 CPU, 16GB Ram, Intel Iris Plus integrated graphics.
Reproducibility: When opening the game exe file the game starts, shows both logos, displays ''loading saves'' and then crashes.

[Player.log](https://github.com/suicvne/DOR_Remake/files/6466788/Player.log)

**Comment by suicvne** [comment]
Ah, the game crashed because it can’t write to the folder “DOR Remake Saves”

`C:\Users\<user>\Documents\DOR Remake Saves`

What are the permissions on the folder? 

I usually develop on macOS where it’s much more stringent on file permissions. The saving/loading was tested against that *assuming* (i know) that Windows would be less stringent. 

Another possibility: The game tries to write a file to that directory called “.tmp_test_file”. Windows may not like this. 

I’ll get to testing on Windows!

**Comment by SpacePirate2k** [comment]
After inspecting the Documents folder in ''C:\Users\<user>\Documents'' there was no ''DOR Remake Saves'' folder generated by the game. I went to that same path and created a ''DOR Remake Saves'' folder and now the game loads correctly. The current permissions for the ''Documents'' folder are: 1) System, 2) My User and 3) Administrators.

![DOR Remake testing](https://user-images.githubusercontent.com/84085365/117996172-8b692980-b307-11eb-9a2b-3b6ebd946bca.png)

**Comment by suicvne** [comment]
Can confirm this same bug happens on Linux when loading the game. I've re-added the code for creating the folder if it doesn't exist (oh the joys of game development!!!!)

While I was doing this, I discovered that Linux Mono does not like to allow you to write files that start with . so I've changed the test file to be written as "tmp_test_file" and deleted after directory validation. (Fixed)

In addition, I managed to discover another bug where I wasn't reading the revision number of the client on a save read. So, if you made a save with 0.6.8.1 it would think that it was made with 0.6.8.0 and mark it as an older save. (Fixed, also)

BTW, if the game is marking your save as "Saved with an older version", the game will not prevent you from loading it. I need to change the message displayed and possibly add a warning when loading, but it's mostly just to let you know that some changes *may* be made to this save making it incompatible with older versions of the game.

Anyway, all these changes will be fixed and all 3 version will be rebuilt against version 0.6.8.1. This issue will be closed once those versions are posted to itch.io. Thanks again for reporting!
